x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/page.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState } from 'react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useUrlState } from '../../../util/url_state';
+import { NodeAvailableWarning } from '../../../components/node_available_warning';
+import { SavedObjectsWarning } from '../../../components/saved_objects_warning';
+import { UpgradeWarning } from '../../../components/upgrade';
+import { JobMap } from '../job_map';
+import { HelpMenu } from '../../../components/help_menu';
+import { useMlKibana } from '../../../contexts/kibana';
+import { useRefreshAnalyticsList } from '../../common';
+import { MlPageHeader } from '../../../components/page_header';
+import { AnalyticsIdSelector } from '../components/analytics_selector';
+
+export const Page: FC = () => {
+  const [globalState] = useUrlState('_g');
+  const [isLoading, setIsLoading] = useState(false);
+  const { refresh } = useRefreshAnalyticsList({ isLoading: setIsLoading });
+  const mapJobId = globalState?.ml?.jobId;
+  const mapModelId = globalState?.ml?.modelId;
+  const [analyticsId, setAnalyticsId] = useState<{ model_id?: string; job_id?: string }>();
```

Comment:
I reckon we should avoid using an intermediate state. both job and model ids should be resolved from the URL state.

Reply:
Storing these in the url so we can do that has been added as a follow up 👍